docs/radio.rst

```diff
@@ -0,0 +1,129 @@
+Radio
+*****
+
+.. py:module:: radio
+
+The ``radio`` module allows devices to work together via simple wireless
+networks.
+
+The radio module is conceptually very simple:
+
+* Broadcast messages are of a certain configurable length (up to 251 bytes).
+* Messages received are read from a queue of configurable size (the larger the queue the more RAM is used). If the queue is full, new messages are ignored.
+* Messages are broadcast and received on a preselected channel (there are 100 available channels to choose from).
+* Broadcasts are at a certain level of power - more power means more range.
+* Messages are filtered by address (like a house number) and group (like a named recipient at the specified address).
+* The rate of throughput can be one of three pre-determined settings.
+* Send and receieve bytes to send arbitrary data.
+* As a convenience for children, it's easy to send and receive messages as strings.
+* The default configuration is both sensible and compatible with other platforms that target the BBC micro:bit.
+
+To access this module you need to::
+
+    import radio
+
+We assume you have done this for the examples below.
+
+Constants
+=========
+
+.. py:attribute:: RATE_250KBIT
+
+    Constant used to indicate a throughput of 256 Kbit a second.
+
+.. py:attribute:: RATE_1MBIT
+
+    Constant used to indicate a throughput of 1 MBit a second.
+
+.. py:attribute:: RATE_2MBIT
+
+    Constant used to indicate a throughput of 2 MBit a second.
+
+
+Functions
+=========
+
+.. py:function:: on()
+
+    Turns the radio on. This needs to be explicitly called since the radio
+    draws power and takes up memory that you may otherwise need.
+
+.. py:function:: off()
+
+    Turns off the radio, thus saving power and memory.
+
+.. py:function:: config(length=32, queue=3, channel=7, power=0, address=0x75626974, group=0, data_rate=radio.RATE_1MBIT)
```

Comment:
Technically the type signature is more like `config(**kwargs)` (since that's how you'd implement this function in Python), where kwargs can be on of the listed parameters.  I'd change it to that, provide a short description what the notation **kwargs means, and then write the default value in the description of each param below.